chain/chain/src/update_shard.rs

```diff
@@ -92,6 +92,7 @@ pub(crate) struct StateSplitData {
 
 /// Reason to update a shard when new block appears on chain.
 /// All types include state roots for split shards in case of resharding.
+#[allow(clippy::large_enum_variant)]
```

Comment:
Out of the scope of the PR: should we consider in future putting the large variants behind a box?  If yes, then not sure if it is worth creating github issues or leaving a todo.

Reply:
This specific case ended up being bogus. Yes, the enum variants are large, but *all* of them are large (and the difference between variants is negligible). So the right way about mitigating any potential performance problems here is to box or otherwise indirect the whole enum, not the individual variants.

(That said, clippy did not flag the relevant parts of the codebase again; :shrug:. Perhaps you’re right that this wants a TODO comment and an issue.)